assets/js/base/components/payment-methods/payment-method-error-boundary.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from 'react';
+import { Notice } from 'wordpress-components';
+import PropTypes from 'prop-types';
+import { CURRENT_USER_IS_ADMIN } from '@woocommerce/block-settings';
+
+class PaymentMethodErrorBoundary extends Component {
+	state = { errorMessage: '', hasError: false };
+
+	static getDerivedStateFromError( error ) {
+		return {
+			errorMessage: error.message,
+			hasError: true,
+		};
+	}
+
+	render() {
+		const { hasError, errorMessage } = this.state;
+		const { isEditor } = this.props;
+
+		if ( hasError ) {
+			return (
+				<Notice isDismissible={ false } status="error">
+					{ errorMessage && ( isEditor || CURRENT_USER_IS_ADMIN )
+						? errorMessage
+						: __(
+								'There was an error loading the payment method.',
```

Comment:
I think this notice appearing to shoppers could be improved a bit, maybe something like this?:

> This site is experiencing difficulties with this payment method. Please contact the owner of the site for assistance.

At some point this may be something we'd want to introduce as a configurable text string.

Reply:
Added in 0814de191360992cf2af7dbc485d097c8392a8d4.